common/djangoapps/student/forms.py

```diff
@@ -144,6 +148,13 @@ class AccountCreationForm(forms.Form):
             "min_length": _PASSWORD_INVALID_MSG,
         }
     )
+    password_copy = forms.CharField(
```

Comment:
@lenacom I think it is very odd to pass the copy of the password to the server-- there's a lot of unnecessary code, and I don't like the user_api being extended to add a password copy field (which doesn't make sense as a property of the user). The front-end check would be very simple-- just checking for equality on two text fields. If it is worth sharing JS related to it, a utility method could be created.

All that being said, I am not the code-reviewer of this PR. I have routed it to the ECOM team because they own these views. We also should have product verify that this is a change we want on the edX site (FYI @marcotuts). @rlucioni can you help move this PR along, including the decision of where the validation should occur (assuming product signs off on this)?